### PR TITLE
Squad Team Leader

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

Read changelog.

# Explain why it's good for the game

RTO is a remnant of an old design and awkwardly sits as assistant squad lead as well as requiring you to take a phone backpack and also ordnance and engineering. It's just sort of bloated and all over the place. Separating the telephone backpack and the idea of a team leader/aSL seems like a prudent move that gives the role a more focused purpose and idea while also allowing for a more generalized spread of radio backpacks across the marine force.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow, Frans_Feiffer
add: Squad RTOs are now Fireteam Leaders
add: Radio telephones can be bought by fireteam leaders or squad leaders
add: Radio telephones can be bought by riflemen
add: Small radio telephones are now universally replaced by the regular version and can be worn with any uniform
fix: Fixed RTO Backpack in hand on_mobs backwards when facing south.
/:cl:

### CODE REVIEW

**Comment by UntoldTactics** [comment]
I'd argue this is a bit superflous. The RTO is a very valuable, useful role and fits rather well, opposed to awkwardly. Their engineering skill tied in with communications even when comms are down makes any RTO valuable for fire support and mortars. Hell, the RTOs themselves (such as in bravo) can also act as engineers + man the mortars, not only freeing up specialized engineers but also being able to use their phone backpack to be called directly to get cords etc. 

They fill a niche in that they're dedicated roles for fire support and communication, allowing CAS and mortar to be more effective than without, even during comms-blackouts. 

It's arguably more bloaty to add a role that's just a fireteam leader, we don't need such large hierarchies.

**Comment by N8-Toe** [comment]
Agreed with untold. This is more bloaty as well the RTO is a very unique communications and support focused role. To loose it would be a shame.

**Comment by frozentsbgg** [comment]
I'm a big fan of this role segmentation of RTO and I mostly like this PR.
That being said I'm really not a fan of the naming of the role, and I can see it being a bit confusing 
74f996801dc3cb0a2f160c9691f6c371-png

Calling it Fire Team leader might be a better name - or just keep RTOs as a throwback. I prefer the later option but I can understand if we want a different name.

**Comment by Diegoflores31** [comment]
Squad Team leader name is kinda confusing with squad leader . it should be just "Team leader" or something like that

**Comment by 4hands44** [comment]
I like the idea but why not give it either the Real world name of Fire team lead, or Assistant SL  The current name is wordy and weird

File: code/game/jobs/job/marine/squad/tl.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	total_positions = 8
+	spawn_positions = 8
+	allow_additional = 1
+	flags_startup_parameters = ROLE_ADD_TO_DEFAULT|ROLE_ADD_TO_SQUAD
+	gear_preset = /datum/equipment_preset/uscm/tl
+	entry_message_body = "You are the <a href='"+URL_WIKI_TL_GUIDE+"'>Team Leader.</a>Your task is to assist the squad leader in leading the squad as well as utilize ordnance such as orbital bombardments, CAS, and mortar as well as coordinating resupply with Requisitions and CIC. If the squad leader dies, you are expected to lead in their place."
+
+/datum/job/marine/tl/generate_entry_conditions(mob/living/carbon/human/H)
+	. = ..()
+	H.important_radio_channels += JTAC_FREQ
```
Comment (harryob) [inline code/game/jobs/job/marine/squad/tl.dm:12]:
```suggestion
	spawning_human.important_radio_channels += JTAC_FREQ
```

File: code/game/jobs/job/marine/squad/tl.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/datum/job/marine/tl
+	title = JOB_SQUAD_TEAM_LEADER
+	total_positions = 8
+	spawn_positions = 8
+	allow_additional = 1
+	flags_startup_parameters = ROLE_ADD_TO_DEFAULT|ROLE_ADD_TO_SQUAD
+	gear_preset = /datum/equipment_preset/uscm/tl
+	entry_message_body = "You are the <a href='"+URL_WIKI_TL_GUIDE+"'>Team Leader.</a>Your task is to assist the squad leader in leading the squad as well as utilize ordnance such as orbital bombardments, CAS, and mortar as well as coordinating resupply with Requisitions and CIC. If the squad leader dies, you are expected to lead in their place."
+
+/datum/job/marine/tl/generate_entry_conditions(mob/living/carbon/human/H)
```
Comment (harryob) [inline code/game/jobs/job/marine/squad/tl.dm:10]:
```suggestion
/datum/job/marine/tl/generate_entry_conditions(mob/living/carbon/human/spawning_human)
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]
i promise to do some fun stuff with fireteams eventually

**Pull request merged**